book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -421,22 +668,24 @@ Returns
 
 Implemented by querying:
 
-- (non-finalized) XXX parts of the non-finalized state;
+- (non-finalized) the `height_by_hash` of each chain starting with the best
+  chain, then find block in `blocks` of that chain.
 - (finalized) the `height_by_hash` (to get the block height) and then
     `block_by_height` (to get the block data) trees.
 
 
 # Drawbacks
 [drawbacks]: #drawbacks
 
-# Rationale and alternatives
-[rationale-and-alternatives]: #rationale-and-alternatives
+- Restarts can cause `zebrad` to redownload up to the last one hundred blocks
+  it verified.
 
-# Prior art
-[prior-art]: #prior-art
+- The service interface puts some extra responsibility on callers to ensure
+  it is used correctly and does not verify the usage is correct at compile
+  time.
 
-# Unresolved questions
-[unresolved-questions]: #unresolved-questions
+- the service API is verbose and requires manually unwrapping enums
 
-# Future possibilities
-[future-possibilities]: #future-possibilities
+- We do not handle reorgs the same way zcashd does, and could in theory need
+  to delete our entire on disk state and resync the chain in some
+  pathological reorg cases.
```

Comment:
I think it would be useful to spell out the pathological case here: it's one where there are multiple chain reorganizations, each of which rewinds less than 100 blocks of state but which rewind more than 100 blocks of state in total.  But this basically requires a 51% attack on the Zcash network, so at that point there are worse problems than having to wipe the block cache and resync.